### ISSUE TITLE
ty 0.0.36 causes apparent regression with Django `IntegerChoices`

### ISSUE THREAD

**Issue by drewbrew** [astral-sh/ty#3468]
### Summary

Referred PR: https://github.com/astral-sh/ruff/pull/25136

Example code on Django 5.2.14:

```python
from django.db import models


class MyModelChoices(models.IntegerChoices):
    GOOD = 1, "I like this"
    LESS_GOOD = 2, "not so much"
    SURE_WHATEVER = 42, "the meaning of life"
```

Behavior on ty 0.0.35 via `ty check`: all checks passed
Behavior on ty 0.0.36:
```
error[invalid-assignment]: Enum member `GOOD` is incompatible with `__new__`
 --> models.py:5:5
  |
5 |     GOOD = 1, "I like this"
  |     ^^^^
  |
info: Expected compatible arguments for `def __new__[Self](cls, value: int) -> Self`

error[invalid-assignment]: Enum member `LESS_GOOD` is incompatible with `__new__`
 --> models.py:6:5
  |
6 |     LESS_GOOD = 2, "not so much"
  |     ^^^^^^^^^
  |
info: Expected compatible arguments for `def __new__[Self](cls, value: int) -> Self`

error[invalid-assignment]: Enum member `SURE_WHATEVER` is incompatible with `__new__`
 --> models.py:7:5
  |
7 |     SURE_WHATEVER = 42, "the meaning of life"
  |     ^^^^^^^^^^^^^
  |
info: Expected compatible arguments for `def __new__[Self](cls, value: int) -> Self`

error[invalid-assignment]: Enum member `GOOD` is incompatible with `__new__`
   --> noun_project/submissions/models.py:163:5
    |
163 |     GOOD = 1, "I like this"
    |     ^^^^
    |
info: Expected compatible arguments for `def __new__[Self](cls, value: int) -> Self`

error[invalid-assignment]: Enum member `LESS_GOOD` is incompatible with `__new__`
   --> noun_project/submissions/models.py:164:5
    |
164 |     LESS_GOOD = 2, "not so much"
    |     ^^^^^^^^^
    |
info: Expected compatible arguments for `def __new__[Self](cls, value: int) -> Self`

error[invalid-assignment]: Enum member `SURE_WHATEVER` is incompatible with `__new__`
   --> noun_project/submissions/models.py:165:5
    |
165 |     SURE_WHATEVER = 42, "the meaning of life"
    |     ^^^^^^^^^^^^^
    |
info: Expected compatible arguments for `def __new__[Self](cls, value: int) -> Self`

Found 6 diagnostics
```

This is an expected use case for these `IntEnum` subclasses (see the last example in the [Enumeration types](https://docs.djangoproject.com/en/6.0/ref/models/fields/#enumeration-types) section of the Django docs).

### Version

ty 0.0.36

**Comment by charliermarsh** [comment]
Perhaps what I added in https://github.com/astral-sh/ruff/pull/25136 is overly aggressive (or lacking a nuance). I'll take a look.

**Comment by charliermarsh** [comment]
`IntegerChoices` has a metaclass with a custom `__new__` that transforms these: https://github.com/django/django/blob/da6567da7ea0e844c9a880e02c7b34e1a184edf8/django/db/models/enums.py#L10

**Comment by drewbrew** [comment]
Yep, I figured there was some Django magic under the hood that your changes didn't expect to see.

**Comment by boxed** [comment]
@drewbrew I've found there's a bunch of Django-isms that ty either doesn't (and probably shouldn't!) know about and that it warns incorrectly for. I'm working on https://github.com/boxed/iommi_lsp to make a Django specific LSP that proxies ty. I pushed a fix for this specific issue just now.

**Comment by drewbrew** [comment]
Thanks, @boxed! I'll have to take a look at that.

**Comment by drewbrew** [comment]
Confirmed, the fix works for me. Thanks so much!
